### PR TITLE
Raise Repair Form

### PR DESCRIPTION
**Pull request by benbaumann95**
### Description of change

**Raise repair form**
- Look at commits for more detail

### Story Link

https://trello.com/c/R4Bj1aFW/279-as-a-user-i-need-to-be-able-to-select-an-sor-code-from-a-list-so-that-the-contractor-know-what-the-job-entails
https://trello.com/c/IG0BVLry/277-as-a-user-i-need-to-be-able-to-enter-additional-description-to-the-repair-so-that-contractors-have-a-better-idea-of-what-needs-t
https://trello.com/c/fHHnSJXg/280-as-a-user-i-need-to-be-able-to-raise-a-works-order-so-that-contractors-know-what-to-fix

### **When you visit the raise repair form page**

Screenshot 2021-01-15 at 15 30 38

### **Filling in fields (auto-populates summary description of SOR code and task priority)**

Screenshot 2021-01-15 at 15 30 59

### **Submitting form without required inputs - error validations**
Screenshot 2021-01-15 at 15 31 19

### **Exceeding 250 max characters for repair description**
Screenshot 2021-01-15 at 15 31 48

### **Successfully submitting the form**
Screenshot 2021-01-15 at 15 32 14

### CODE REVIEW

File: src/pages/api/raise-repair.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+export default async (req, res) => {
+  try {
+    const { status, data } = await postRaiseRepairForm(req.body)
+
+    res.status(status).json(data)
+  } catch (err) {
+    console.log(err)
+    return { status: 500 }
+  }
+}
```
Comment (EGiataganas) [inline src/pages/api/raise-repair.js:12]: Could you please update it to use Authorisation and update it to use `http-status-codes`?

File: cypress/integration/raise-repair/form.spec.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  cy.route('GET', 'api/sor-codes', '@property')
+  cy.route('GET', 'api/sor-codes', '@sorCodes')
+  cy.route({
+    method: 'POST',
+    url: '/api/raise-repair',
+    response: '10102030',
+  }).as('apiCheck')
+})
+
+describe('Raise repair form', () => {
```
Comment (NeilvB) [inline cypress/integration/raise-repair/form.spec.js:29]: This is a super-readable integration test

**Review by NeilvB — COMMENTED** [review]
This is a lot nice work!

Only thing I think we need to add is authorisation on the API route endpoints, as per our existing ones.

As a non-blocking minor question, is it possible to create a component for constructing the HTML in [this function](https://github.com/LBHackney-IT/repairs-hub-frontend/pull/32/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.lock&file-filters%5B%5D=.scss&file-filters%5B%5D=dotfile#diff-cfe07f7e1ee7795417fcbe8183a9d445a53bd6285cabb3816655f6695134877bR58)?

File: src/components/Form/Select/Select.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import ErrorMessage from '../../Errors/ErrorMessage/ErrorMessage'
```
Comment (siapankina) [inline src/components/Form/Select/Select.js:3]: just checked what this `"classnames"` package is doing, really nice to have it here

**Comment by benbaumann95** [comment]
> This is a lot of nice work! 💯
> 
> Only thing I think we need to add is authorisation on the API route endpoints, as per our existing ones.
> 
> As a non-blocking minor question, is it possible to create a component for constructing the HTML in [this function](https://github.com/LBHackney-IT/repairs-hub-frontend/pull/32/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.lock&file-filters%5B%5D=.scss&file-filters%5B%5D=dotfile#diff-cfe07f7e1ee7795417fcbe8183a9d445a53bd6285cabb3816655f6695134877bR58)?

Thanks Neil. I think you probably could extract it into a component with some tweaking but don't believe you can render a component directly from the function i've made. Not sure if it's overkill for this or not but happy to add a chore to do this if it would be easier to manage/read.

**Review by pixeltrix — APPROVED** [review]
:shipit:

**Review by NeilvB — APPROVED** [review]

**Pull request merged**